specs/altair/beacon-chain.md

```diff
@@ -511,16 +511,28 @@ def apply_deposit(state: BeaconState,
         signing_root = compute_signing_root(deposit_message, domain)
         # Initialize validator if the deposit signature is valid
         if bls.Verify(pubkey, signing_root, signature):
-            state.validators.append(get_validator_from_deposit(pubkey, withdrawal_credentials, amount))
-            state.balances.append(amount)
+            index = assign_index_to_deposit(state)
+            update_list(state.validators, index, get_validator_from_deposit(pubkey, withdrawal_credentials, amount))
+            update_list(state.balances, index, amount)
```

Comment:
I see the code-reuse niceness of creating `update_list()`. It's a bit indirect for our standard use. Maybe a name-change would help -- `update_or_append_to_list()` to be more explicit about what's happening